packages/task/src/node/process/process-task.ts

```diff
@@ -70,7 +70,7 @@ export class ProcessTask extends Task {
         });
     }
 
-    getRuntimeInfo(): ProcessTaskInfo {
+    async getRuntimeInfo(): Promise<ProcessTaskInfo> {
```

Comment:
this function does not need to be `async` i guess ?

Reply:
ahh, i see this line: https://github.com/theia-ide/theia/blob/master/packages/task/src/node/task-runner.ts#L32

just thinking: maybe we should change the return type of the function in the interface from `Promise<...>` to `MaybePromise<...>` ?

Reply:
fixed

Reply:
@elaihau could you please review my fix

Reply:
@vinokurig sorry i still don't understand why updating `getRuntimeInfo()` is related to https://github.com/eclipse/che-theia-task-plugin/blob/master/che-theia-task-extension/src/node/che-task-runner.ts#L21

Could you please kindly explain ?

Reply:
I need it for my PR: https://github.com/eclipse/che-theia/blob/09515ef56850be2186afa23ba1188945682f5a72/extensions/eclipse-che-theia-plugin-ext/src/node/task-service.ts#L77

Reply:
Got it. I am not an expert of `Promise` - if I were you, I would update
[this line](https://github.com/eclipse/che-theia/blob/09515ef56850be2186afa23ba1188945682f5a72/extensions/eclipse-che-theia-plugin-ext/src/node/task-service.ts#L77) in your other PR into:
```ts
async getRuntimeInfo(): Promise<TaskInfo> {
```
and revert the change made to [this line](https://github.com/theia-ide/theia/pull/3909/files#diff-bccc335bccb5d8638a4987cf68ee36b3R73) in this PR.

You could check if what I suggested works.

Reply:
fixed
@elaihau Thank you for your advise, it works.